src/mlpack/tests/saga_test.cpp

```diff
@@ -65,10 +75,16 @@ BOOST_AUTO_TEST_CASE(GeneralizedRosenbrockTest)
     arma::mat coordinates = f.GetInitialPoint();
     double result = optimizer.Optimize(f, coordinates);
 
-    BOOST_REQUIRE_SMALL(result, 1e-10);
     for (size_t j = 0; j < i; ++j)
-      BOOST_REQUIRE_CLOSE(coordinates[j], (double) 1.0, 1e-3);
+      if (abs(coordinates[j] - 1.0) < 1e-3)
+        indicator++;
+    if (result< 1e-10 && indicator == i)
+    {
+      success++;
+      break;
```

Comment:
Maybe I missed something, but it looks like if the condition is true this will break, and we don't test further batches.

Reply:
I think since we're testing that SAGA converges on atleast one variant of `Rosenbrock`function then there's no point to run the loop further. Same has been done in this https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/recurrent_network_test.cpp#L83

Reply:
The recurrent network test is different, we only test a single setting, in this case we'd like to know if SAGA can handle different batch sizes, but we will stop after the first one (if successful).

Reply:
@zoq I've modified the Test for testing batches. Is this the correct way to test batches? But still SAGA failed one time out of three on Travis. I couldn't even reproduce the results as the Test will always pass on my system. I couldn't figure it out what's wrong. I've again thoroughly checked the algorithm implemented in SAGA but I find it correct. I tried the implementation in python and applied to a dataset the results were pretty good. What do you think is going wrong?